app/controllers/CacheController.scala

```diff
@@ -153,8 +159,10 @@ class CacheController @Inject() (
         cacheRepository
           .get(lrn, eoriNumber)
           .map {
-            case Some(userAnswers) =>
+            case Some(userAnswers) if !requestHasPhase() || isRequestTransitional() == userAnswers.isTransitional =>
```

Comment:
I see what you've done here. You've tried to make it compatible with the frontends as they currently are as they don't attach the header to the request. However, this creates a loophole, in that one could regain access to their answers by omitting the header.

What would be preferable I think would be to make use of the getVersion action builder, with the relevant changes in the various frontends. When it comes to deployments, it would be imperative that the frontends get deployed first.

Reply:
In terms of the frontend changes, we can build the view once in manage-transit-movements-departure-frontend, and redirect to that URL from the other smaller departure frontends (similar to what we do for error pages etc.)